samza-api/src/main/java/org/apache/samza/system/SystemStreamPartition.java

```diff
@@ -74,6 +94,7 @@ private int computeHashCode() {
     final int prime = 31;
     int result = super.hashCode();
     result = prime * result + ((partition == null) ? 0 : partition.hashCode());
+    result = prime * result + ((keyBucket == -1) ? 0 : keyBucket);
```

Comment:
I just realized this hashing function yields the same hash regardless of elasticity enabled or not for bucket 0 vs -1 respectively.

I feel you could treat 0 as default and modify the `toString()` to include `keyBucket` regardless and just regenerate dashboard or make this is API break change in metrics name explicitly as we are anyways going to run into it.